modules/core/src/main/scala/com/azavea/stac4s/core/StacLinkType.scala

```diff
@@ -1,61 +1,69 @@
 package com.azavea.stac4s
 
+import cats.Eq
 import cats.implicits._
 import io.circe._
 
 sealed abstract class StacLinkType(val repr: String) {
   override def toString = repr
 }
-case object Self        extends StacLinkType("self")
-case object StacRoot    extends StacLinkType("root")
-case object Parent      extends StacLinkType("parent")
-case object Child       extends StacLinkType("child")
-case object Item        extends StacLinkType("item")
-case object Items       extends StacLinkType("items")
-case object Source      extends StacLinkType("source")
-case object Collection  extends StacLinkType("collection")
-case object License     extends StacLinkType("license")
-case object Alternate   extends StacLinkType("alternate")
-case object DescribedBy extends StacLinkType("describedBy")
-case object Next        extends StacLinkType("next")
-case object Prev        extends StacLinkType("prev")
-case object ServiceDesc extends StacLinkType("service-desc")
-case object ServiceDoc  extends StacLinkType("service-doc")
-case object Conformance extends StacLinkType("conformance")
-case object Data        extends StacLinkType("data")
-
-final case class VendorLinkType(underlying: String) extends StacLinkType("vendor") {
-  override def toString = s"$repr-$underlying"
-}
 
 object StacLinkType {
 
+  implicit def eqStacLinkType: Eq[StacLinkType] = Eq[String].imap(fromString _)(_.repr)
+
+  case object Self               extends StacLinkType("self")
+  case object StacRoot           extends StacLinkType("root")
+  case object Parent             extends StacLinkType("parent")
+  case object Child              extends StacLinkType("child")
+  case object Item               extends StacLinkType("item")
+  case object Items              extends StacLinkType("items")
+  case object Source             extends StacLinkType("source")
+  case object Collection         extends StacLinkType("collection")
+  case object License            extends StacLinkType("license")
+  case object Alternate          extends StacLinkType("alternate")
+  case object DescribedBy        extends StacLinkType("describedBy")
+  case object Next               extends StacLinkType("next")
+  case object Prev               extends StacLinkType("prev")
+  case object ServiceDesc        extends StacLinkType("service-desc")
+  case object ServiceDoc         extends StacLinkType("service-doc")
+  case object Conformance        extends StacLinkType("conformance")
+  case object Data               extends StacLinkType("data")
```

Comment:
These moved inside the object because `Data` is both a link type and an asset role and they need not to be in the same namespace